src/equations/shallow_water_two_layer_1d.jl

```diff
@@ -0,0 +1,586 @@
+  [DOI: 10.1142/9789814417099_0039](https://doi.org/10.1142/9789814417099_0039)
+It should be noted that the equations are ordered differently and the
+designation of the upper and lower layer has been changed which leads to a slightly different
+formulation.
+"""
+@inline function flux_es(u_ll, u_rr,
+                         orientation::Integer,
+                         equations::ShallowWaterTwoLayerEquations1D)
+  # Compute entropy conservative flux but without the bottom topography
+  f_ec = flux_fjordholm_etal(u_ll, u_rr,
+                              orientation,
+                              equations)[1:4]
+
+  # Get maximum signal velocity
+  λ = max_abs_speed_naive(u_ll, u_rr, orientation, equations)
+
+  # Get entropy variables but without the bottom topography
+  q_rr = cons2entropy(u_rr,equations)[1:4]
+  q_ll = cons2entropy(u_ll,equations)[1:4]
+
+  # Average values from left and right
+  u_avg = (u_ll + u_rr) / 2
+
+  # Introduce variables for better readability
+  rho1 = equations.rho1
+  rho2 = equations.rho2
+  g    = equations.gravity
+  drho = rho1 - rho2
+
+  # Entropy Jacobian matrix
+  H = Array{Float64,2}(undef,4,4)
+  # Use symmetry properties to compute matrix
+  H[:,1] = 
+    [-rho2/(g*rho1*drho);;
+    -rho2*u_avg[2]/(g*rho1*u_avg[1]*drho);;
+      1.0/(g*drho);;
+      u_avg[4]/(g*u_avg[3]*drho)]
+  H[:,2] = 
+    [H[2,1];;
+      (g*rho1*u_avg[1]^3 - g*rho2*u_avg[1]^3 - rho2*u_avg[2]^2)/(g*rho1*u_avg[1]^2*drho);;
+      u_avg[2]/(g*u_avg[1]*drho);;
+      u_avg[2]*u_avg[4]/(g*u_avg[1]*u_avg[3]*drho)]
+  H[:,3] =
+    [H[3,1];;
+      H[3,2];;
+    -1.0/(g*drho);;
+    -u_avg[4]/(g*u_avg[3]*drho)]
+  H[:,4] = 
+    [H[4,1];;
+      H[4,2];;
+      H[4,3];;
+      (g*rho1*u_avg[3]^3 - g*rho2*u_avg[3]^3 - rho2*u_avg[4]^2)/(g*rho2*u_avg[3]^2*drho)]
+
+  # Add dissipation to entropy conservative flux to obtain entropy stable flux
+  f_es = f_ec - 0.5 * λ * H * (q_rr - q_ll)
+
+  return SVector(f_es[1], f_es[2], f_es[3], f_es[4], zero(eltype(u_ll)))
+end
+
+
+# Calculate approximation for maximum wave speed for local Lax-Friedrichs-type dissipation as the
+# maximum velocity magnitude plus the maximum speed of sound. This function uses approximate 
+# eigenvalues using the speed of the barotropic mode as there is no simple way to calculate them 
+# analytically.
+# 
+# A good overview of the derivation is given in:
+# -  Jonas Nycander, Andrew McC. Hogg, Leela M. Frankcombe (2008)
+#    Open boundary conditions for nonlinear channel Flows
+#    [DOI: 10.1016/j.ocemod.2008.06.003](https://doi.org/10.1016/j.ocemod.2008.06.003)
+@inline function max_abs_speed_naive(u_ll, u_rr,
+                                     orientation::Integer,
+                                     equations::ShallowWaterTwoLayerEquations1D)
+  # Get the averaged velocity
+  Um_ll = (u_ll[2] + u_ll[4]) / (u_ll[1] + u_ll[3])
+  Um_rr = (u_rr[2] + u_rr[4]) / (u_rr[1] + u_rr[3])
```

Comment:
We usually use `v` to denote velocities

Reply:
Changed variable names to `v_m_ll` and `v_m_rr` in 7c9fde8356a8fc13cdea0b9313579ed924402a41